### PR TITLE
Added tests for /monitor/metrics/{metrics_type}     

### LINKED ISSUE #587: Tests for /monitor/metrics/<name> endpoint
Write tests for the change added with https://github.com/ipfs/ipfs-cluster/pull/572
Started with https://github.com/ipfs/ipfs-cluster/pull/572#issuecomment-432046084

### PR DESCRIPTION
**Pull request by kishansagathiya**
Added API, client tests and sharness tests for GET /monitor/metrics/{metrics_type}
    
Fixes #587

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 684550694194a6c0e91b8793bc2b96898ef16bf6
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by:**

 * Commit: 5df53406b5a70ef65f239674acece77b445e7c47
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by:**

 * Commit: af7f03150241a4c5b4dd30596119b0abada9bfb6
  * Invalid signoff. Commit message must end with  
    **License: MIT  
Signed-off-by:**

Code contribution guidelines for IPFS Cluster are available at https://cluster.ipfs.io/developer/contribute/#code-contribution-guidelines

---
This message was auto-generated by https://gitcop.com

**Comment by kishansagathiya** [comment]
@hsanjuan
Haven't tested this manually, yet(writing a script to make manual testing easier for me), hopefully that won't be a problem as this PR addresses only tests.

File: api/rest/restapi_test.go [hunk 1]
```diff
@@ -556,6 +556,30 @@ func TestAPIAllocationEndpoint(t *testing.T) {
...
+				}
+				if m.Peer != test.TestPeerID1.Pretty() {
+					t.Error("Unexpected peer id: ", m.Peer)
+				}
+			}
+		}
+	}
+
+	testBothEndpoints(t, tf)
+}
```
[the remark this answers is not in the document]
Comment (kishansagathiya) [reply api/rest/restapi_test.go]: done in the next commit.

**Comment by kishansagathiya** [comment]
@hsanjuan Modified as changes requested

It's about time I start pulling new changes instead of rebasing, to avoid force-push :laughing:

**Comment by lanzafame** [comment]
@kishansagathiya rebasing is fine. You aren't force pushing to master, so it is fine :)

**Comment by hsanjuan** [comment]
@lanzafame looks good on your side ?

**Pull request merged**